pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1917,7 +1913,27 @@ impl<'a> DatadirModification<'a> {
     }
 
     fn put(&mut self, key: Key, val: Value) {
-        let values = self.pending_updates.entry(key).or_default();
+        if key.is_rel_block_key() || key.is_slru_block_key() {
+            self.put_data(key, val)
+        } else {
+            self.put_metadata(key, val)
+        }
```

Comment:
The distinction between data and metadata is fuzzy for me. Could you lift the "is key data predicate" into a function and add a doc comment which explains the distinction?

Reply:
My confusion comes from the SLRU keys. They are technically metadata, but we treat them as data.

Reply:
Good call: clarified the data/metadata thing in a helper function here: 68a6c46d1